File: Java/0007_reverse_integer/0007_reverse_integer.java
public class Solution {
    public int reverse(int x) {
        int rev = 0;
        while (x != 0) {
            if (rev != 0 && Integer.MAX_VALUE / rev < 10 && Integer.MAX_VALUE / rev > -10)
                return 0;
            rev = rev * 10 + x % 10;
            x = x / 10;
        }
        return rev; 
    }
}

File: Python/0007_reverse_integer/0007_reverse_integer.py
class Solution:
    
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if operator.gt(x, 0):
            n = int(str(abs(x))[::-1])
        else:
            n = -1 * int(str(abs(x))[::-1])
            
        return n if n.bit_length() < 32 else 0
